version.py

```diff
@@ -21,4 +21,4 @@
 # Typically, adding a dependency only requires a minor version bump, and
 # removing a dependency requires a major version bump.
 
-PROVISION_VERSION = '40.0'
+PROVISION_VERSION = '40.1'
```

Comment:
If the new mypy isn’t backwards compatible with the old Zulip code (and, given the other changes, it looks like it’s not), this needs to be a major bump.